### DISCUSSION TITLE
Unable to change machine type

### DISCUSSION THREAD

**Discussion by mrmachine** [community/community#7087] [Codespaces] [answered]
I initially created a 2-core/4GB/32GB codespace, then needed to run a process that needed >8GB. I went to change the machine type, but I could only select between 4GB/8GB (both have 32GB storage).

Then I deleted and created an 8-core/16GB/64GB codespace to run my process. But after having run the process, I tried to change the machine type and there were NO other options. This machine type is the only one with 64GB storage.

Is it the case that you can ONLY switch between machine types that have identical storage capacity?

This GREATLY reduces the benefit in being able to change the machine type. You can switch between 4GB/8GB, but that's it. Can't go bigger, or go from the bigger sizes to any other size.

I do not want to be forever locked into working on an 8-core/16GB Codespace all day long just because I have to run one memory intensive process to setup the Codespace.

Why is storage size coupled to machine type, at all? I am sure many people might like to use 128GB storage on a 4GB Codespace, or 32GB storage on a 32GB Codespace. We should be able to select storage and machine type independently, as both incur costs independently.

**Comment by metaskills** [comment]
I was confused by this as well. I started with a 4 core but then thought later on I needed an 8 core machine since I am running additional services with compose. I would have expected that change machine type would have shown me the same types as the initial create options.

<img width="645" alt="Screen Shot 2021-11-05 at 1 37 20 PM" src="https://user-images.githubusercontent.com/2381/140554534-eb0f7e69-8a38-492b-925f-6451435510bc.png">

<img width="691" alt="Screen Shot 2021-11-05 at 9 30 00 AM" src="https://user-images.githubusercontent.com/2381/140554545-f766d234-95ac-4600-b89d-aba36a403241.png">

**Comment by longzheng** [comment]
I'm also running into this issue where I initially created a "4 core" machine and was hoping to upgrade to the "8 core" but the dialog only gives me "2 core" and "4 core"

![image](https://user-images.githubusercontent.com/484912/148328769-995f808d-96ab-4022-9ace-6a26f519f531.png)

**Reply by squirmlabs** [reply]
I also have a similar issue.

This is what I have performed in hopes to see the larger machine types in the `change codespace machine type` selection:

I have checked:

1. Organization Settings → Codespaces → Policies → Machine types - Included "8-core" or "16-core"
2.  Set budgeting/spending limit to ($100)
3. Repository is in the allowed list 
4. I've tried setting `hostRequirements`

```
 "hostRequirements": {
   "cpus": 16,
   "memory": "32gb",
   "storage": "32gb"
},

"hostRequirements": {
  "cpus": 8,     
  "memory": "16gb",
  "storage": "32gb"
},

```
5. I tried lowering it to 4 cores first to get it working, but I always end up with 

```
./scripts/test-webkit-build.sh
================================
WebKit Build Test for Codespaces
================================
✓ Running in GitHub Codespaces

System Information:
  CPU Cores: 2
  Total RAM: 7.8Gi
  Available RAM: 5.7Gi
  Disk Space: 28G available

Build Configuration:
  Parallel Jobs: 2 (based on 2GB RAM per core)
  Build Type: debug
```
<img width="956" height="713" alt="Screenshot 2025-12-12 at 02 57 03" src="https://github.com/user-attachments/assets/cf940765-4208-4125-bd01-e8cddcf30435" />
<img width="968" height="281" alt="Screenshot 2025-12-12 at 02 57 29" src="https://github.com/user-attachments/assets/3ac288f0-26c7-40e1-aa73-17deb5064a0a" />
<img width="1083" height="190" alt="Screenshot 2025-12-12 at 02 58 37" src="https://github.com/user-attachments/assets/4342d93d-5f9f-4ec8-913f-daaae837f683" />
<img width="1328" height="671" alt="Screenshot 2025-12-12 at 02 59 49" src="https://github.com/user-attachments/assets/1b5feb59-7cb1-4fa8-adca-698b4f0f4006" />


Please help.

**Comment by tanmayeekamath** [comment, marked as answer]
Thanks for your feedback. We currently support switching machine types with the same storage specs. So if you have a 2 core machine (with 32 GB storage) then you can only switch to a 4 core since it has the same storage spec. We will update our docs to reflect the current behavior to avoid future confusion up until we support this scenario. 

We're aware that this is not ideal and have a workitem to support updating machine types across storage types, unfortunately I don't have a timeline for supporting this just yet, but we will update this issue as soon as we have one.

**Reply by huseyincenik** [reply]
I have a Github Student Package account. I can only access 2 core and 4 core machines on my Codespaces screen. Why can't I access other machines? @tanmayeekamath . I want to use other machines too
![image](https://github.com/community/community/assets/127469334/2d0af053-a787-4e7c-9b9c-5f1e85404691)

**Reply by Guy-Lay** [reply]
Yes — currently you can only switch between machine types with the same storage capacity.
This does reduce the practical usefulness of machine-type switching, especially for users with occasional high-RAM workloads.
Many developers (including you) are asking for separate controls for storage and compute so Codespaces can better support variable workloads.

**Reply by Musicminion** [reply]
It’s 2026.8 now — is there any update on this?

Being able to choose the machine type and storage size independently would still be very useful, especially for workloads that need more disk space but not necessarily more CPU or RAM.

**Comment by sagor740** [comment]
B930:1049FE:B550FE:C774D9:6A8DF6A8

**Reply by sagor740** [reply]
// Source - https://stackoverflow.com/q/15794179
// Posted by Hanny, modified by community. See post 'Timeline' for change history
// Retrieved 2026-08-26, License - CC BY-SA 4.0

    $stationFilter = $_GET['station'];
    $verticalFilter = $_GET['vertical'];
    $creativeFilter = $_GET['creative'];
    $weekFilter = $_GET['week'];    
    
    $result = mysql_query("SELECT * FROM tableName WHERE STATION_NETWORK = '$stationFilter' AND VERTICAL = '$verticalFilter' AND CREATIVE = '$creativeFilter'  AND WK = '$weekFilter'");   
    $data = array();
    while ($row = mysql_fetch_row($result) )
        {
        $data[] = $row;
        }   
    $finalarray['rowdata'] = $data;
